optimize/client/src/modules/components/Breadcrumbs/Breadcrumbs.test.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import {runLastEffect} from '__mocks__/react';
+import {Link, useLocation} from 'react-router-dom';
+import {shallow} from 'enzyme';
+import {BreadcrumbItem, BreadcrumbSkeleton} from '@carbon/react';
+
+import Breadcrumbs from './Breadcrumbs';
+import {getEntityId, loadEntitiesNames} from './service';
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: jest.fn(),
+}));
+
+jest.mock('hooks', () => ({
+  useErrorHandling: () => ({
+    mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
+  }),
+}));
+
+jest.mock('./service', () => ({
+  loadEntitiesNames: jest.fn(),
+  getEntityId: jest.fn(),
+}));
+
+const mockUseLocation = useLocation as jest.Mock;
+const mockLoadEntitiesNames = loadEntitiesNames as jest.Mock;
+const mockGetEntityId = getEntityId as jest.Mock;
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+it('should render BreadcrumbSkeleton while entityNames is not set', () => {
```

Comment:
I would recommend against mentioning internal implementations details like `BreadcrumbSkeleton` and `entityNames`. It should be something like `it should render a skeleton` and then the test shows the logic necessary for that to happen so the tests are more resilient to change.

I know it's probably done like this everywhere in the Optimize codebase but we could start improving it little by little.